c++/src/kj/async-inl.h

```diff
@@ -913,6 +940,22 @@ inline ExceptionOrValue& ForkBranchBase::getHubResultRef() {
   return hub->getResultRef();
 }
 
+
+template <typename T>
+class BranchCoroPromise {
+  // coro-only promise-like value object for addCoroBranch() implementation
+
+public:
+  KJ_DISALLOW_COPY(BranchCoroPromise);
```

Comment:
I wish C++ had some feature where you could say "if this value is to be copied or moved, convert it to a different type", so that we could convert to `Promise<T>` in that case. Then we wouldn't need a separate method, we could auto-detect when the caller is going to co_await immediately.

But I don't think it does.

Reply:
Can you give some example code on what you'd want the behavior to be? Move and copy semantics are surprisingly deep (but sometimes the edge cases hurt).